lib/hsa/mcwamp_hsa.cpp

```diff
@@ -323,20 +327,42 @@ class HSAKernel {
         hsaExecutableSymbol(_hsaExecutableSymbol),
         kernelCodeHandle(_kernelCodeHandle) {
 
-            hsa_status_t status = 
-                hsa_executable_symbol_get_info(
-                    _hsaExecutableSymbol,
-                    HSA_EXECUTABLE_SYMBOL_INFO_KERNEL_GROUP_SEGMENT_SIZE,
-                    &this->static_group_segment_size);
-            STATUS_CHECK_Q(status, commandQueue, __LINE__);
+        hsa_status_t status = 
+            hsa_executable_symbol_get_info(
+                _hsaExecutableSymbol,
+                HSA_EXECUTABLE_SYMBOL_INFO_KERNEL_GROUP_SEGMENT_SIZE,
+                &this->static_group_segment_size);
+        STATUS_CHECK_Q(status, commandQueue, __LINE__);
+
+        status = 
+            hsa_executable_symbol_get_info(
+                _hsaExecutableSymbol,
+                HSA_EXECUTABLE_SYMBOL_INFO_KERNEL_PRIVATE_SEGMENT_SIZE,
+                &this->private_segment_size);
+        STATUS_CHECK_Q(status, commandQueue, __LINE__);
+    
+        wavefront_sgpr_count = 0;
+        workitem_vgpr_count = 0;
+
+        bool ext_supported = false;
+        status = hsa_system_extension_supported(HSA_EXTENSION_AMD_LOADER, 1, 0, &ext_supported);
+        STATUS_CHECK_Q(status, commandQueue, __LINE__);
 
-            status = 
-                hsa_executable_symbol_get_info(
-                    _hsaExecutableSymbol,
-                    HSA_EXECUTABLE_SYMBOL_INFO_KERNEL_PRIVATE_SEGMENT_SIZE,
-                    &this->private_segment_size);
+        if (ext_supported) {
```

Comment:
Maybe we'd want to strictly enforcing this since we know we always have this extension on rocm